src/chat.h

```diff
@@ -152,13 +159,13 @@ class Chat
 
 	private:
 		std::map<uint16_t, ChatChannel> normalChannels;
-		std::map<uint16_t, PrivateChatChannel*> privateChannels;
-		std::map<Party*, ChatChannel*> partyChannels;
-		std::map<uint32_t, ChatChannel*> guildChannels;
+		std::map<uint16_t, PrivateChatChannel> privateChannels;
+		std::map<Party*, ChatChannel> partyChannels;
+		std::map<uint32_t, ChatChannel> guildChannels;
 
 		LuaScriptInterface m_scriptInterface;
 
-		ChatChannel* dummyPrivate;
+		std::unique_ptr<PrivateChatChannel> dummyPrivate;
```

Comment:
Actually, it doesn't have to. I'll embed that variable directly into the Chat class.